crates/sim2h/src/lib.rs

```diff
@@ -107,17 +118,18 @@ pub enum DhtAlgorithm {
 
 struct Sim2hState {
     crypto: Box<dyn CryptoSystem>,
-    connection_states: HashMap<Lib3hUri, ConnectionState>,
-    open_connections: HashMap<
-        Lib3hUri,
-        (
-            Arc<Mutex<ConnectionJob>>,
-            crossbeam_channel::Sender<WsFrame>,
-        ),
-    >,
+    connection_states: HashMap<Lib3hUri, ConnectionStateItem>,
+    open_connections: HashMap<Lib3hUri, OpenConnectionItem>,
     spaces: HashMap<SpaceHash, Space>,
 }
 
+type ConnectionStateItem = (String, ConnectionState);
+type OpenConnectionItem = (
+    String, // uuid
+    Arc<Mutex<ConnectionJob>>,
+    crossbeam_channel::Sender<WsFrame>,
+);
```

Comment:
There's an odd implicit assumption in these structs (that also existed before your changes) that one connection will only ever join one space.

- If we're going to continue with that assumption, should we use the same uuid for both of these? + maybe throw an error if we get a second JoinSpace?
- If we want to allow multiple spaces per connection, should we make ConnectionState actually support that?

Reply:
Well, currently that optimization would have to happen across core and sim2h, and would require the conductor to hand the sim2h_worker a sim2h connection, instead of the sim2h_worker initiating that connection.  Given that total connections seems to be one of our bottlenecks we probably ought to fix this, and I'll put it into the tree, but I think we can do that later.